src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -1863,7 +1863,7 @@ class RGWArchiveBucketMetadataHandler : public RGWBucketMetadataHandler {
 
     new_be.bucket.name = new_bucket_name;
 
-    ret = ctl.bucket->store_bucket_instance_info(be.bucket, new_bi, y, dpp, RGWBucketCtl::BucketInstance::PutParams()
+    ret = ctl.bucket->store_bucket_instance_info(new_be.bucket, new_bi, y, dpp, RGWBucketCtl::BucketInstance::PutParams()
```

Comment:
so we were updating the existing bucket instance metadata, then deleting it with `remove_bucket_instance_info()` just below? neither `store_bucket_instance_info()` nor `remove_bucket_instance_info()` delete the bucket index objects, but this removal of the bucket instance metadata would effectively orphan the bucket index shards and all of the object versions they contained

Reply:
since we are only renaming the bucket, is it necessary to clean the bucket index shards? 
I can list the objects and do all sorts of get bucket operations on this renamed bucket.

Reply:
> is it necessary to clean the bucket index shards?

no. you're right, we still need the index